packages/@react-stately/table/src/useTableColumnResizeState.ts

```diff
@@ -0,0 +1,181 @@
+    return isStatic(width) ? {...acc, staticColumns: [...acc.staticColumns, column]} : {...acc, dynamicColumns: [...acc.dynamicColumns, column]}; 
+  }, {staticColumns: [], dynamicColumns: []}), [getRealColumnWidth]);
+
+  let buildColumnWidths = useCallback((affectedColumns: GridNode<T>[], availableSpace: number): Map<Key, number> => {
+    const widths = new Map<Key, number>();
+    let remainingSpace = availableSpace;
+
+    const {staticColumns, dynamicColumns} = getStaticAndDynamicColumns(affectedColumns);
+
+    staticColumns.forEach(column => {
+      let width = getRealColumnWidth(column);
+      let w = parseStaticWidth(width, tableWidth.current);
+      widths.set(column.key, w);
+      remainingSpace -= w;
+    });
+
+    // dynamic columns
+    if (dynamicColumns.length > 0) {
+      const newColumnWidths = getDynamicColumnWidths(dynamicColumns, remainingSpace, tableWidth.current);
+      for (let column of newColumnWidths) {
+        widths.set(column.key, column.calculatedWidth);
+      }
+    }
+
+    return widths;
+  }, [getStaticAndDynamicColumns, getRealColumnWidth]);
+
+  // if the columns change, need to rebuild widths.
+  useLayoutEffect(() => {
+    columnsRef.current = columns;
+    const widths = buildColumnWidths(columns, tableWidth.current);
+    setColumnWidthsForRef(widths);
+  }, [columns, buildColumnWidths]);
+
+  function setColumnWidthsForRef(newWidths: Map<Key, number>) {
+    columnWidthsRef.current = newWidths;
+    // new map so that change detection is triggered
+    setColumnWidths(newWidths);
+  }
+
+  function setTableWidth(width: number) {
+    if (width && width !== tableWidth.current) {
+      tableWidth.current = width;
+      if (!isResizing.current) {
+        const widths = buildColumnWidths(columns, width);
+        setColumnWidthsForRef(widths);
+      }
+    }
+  }
+
+  function onColumnResizeStart() {
+    isResizing.current = true;
+    startResizeContentWidth.current = getContentWidth(columnWidthsRef.current);
+  }
+
+  function onColumnResize(column: GridNode<T>, width: number) {
+    let widthsObj = resizeColumn(column, width);
+    props.onColumnResize && props.onColumnResize(widthsObj);
+  }
+
+  function onColumnResizeEnd(column: GridNode<T>, width: number) {
+    isResizing.current = false;
+    if (props.onColumnResizeEnd) {
+      let widthsObj = resizeColumn(column, width);
+      props.onColumnResizeEnd(widthsObj);
+    }
+
+    let widths = new Map<Key, number>(columnWidthsRef.current);
+    widths.set(columnsRef.current[columnsRef.current.length - 1].key, 0);
+    setColumnWidthsForRef(widths);
+  }
+
+  function resizeColumn(column: GridNode<T>, newWidth: number) : { key: Key, width: number }[] {
+    // copy the columnWidths map and set the new width for the column being resized
+    let widths = new Map<Key, number>(columnWidthsRef.current);
+    widths.set(columnsRef.current[columnsRef.current.length - 1].key, 0);
+    widths.set(column.key, Math.max(
```

Comment:
I don't understand this double widths.set, seems to be using the spooky column? but shouldn't that already be at 0?

Reply:
@snowystinger yes you're right about the spooky column. @marshallpete could you add a comment here in the code explaining it? It might be helpful to pull out the `columnsRef.current[columnsRef.current.length - 1].key` into a variable so we know exactly what is being set to 0 and why.

Reply:
explanation in the code: [77c0b56](https://github.com/adobe/react-spectrum/pull/2883/commits/77c0b564176687c1574f44b6695e4167a3a451a7)